sammi/cdf_attribute_manager.py

```diff
@@ -197,24 +356,91 @@ def get_global_attributes(self, instrument_id: str | None = None) -> dict:
             elif attr_schema["required"] and attr_name not in self._global_attributes:
                 # TODO throw an error
                 output[attr_name] = None
-
         return output
 
-    def load_variable_attributes(self, file_name: str) -> None:
+    def global_attribute_template(self) -> OrderedDict:
         """
-        Add variable attributes for a given filename.
+        Function to generate a template of required global attributes
+        that must be set for a valid data file.
+
+        Returns
+        -------
+        template : `OrderedDict`
+            A template for required global attributes that must be provided.
+        """
+        template = OrderedDict()
+        for attr_name, attr_schema in self.global_attribute_schema.items():
+            if attr_schema["required"] and attr_name not in self._global_attributes:
+                template[attr_name] = None
+        return template
+
+    def global_attribute_info(self, attribute_name: Optional[str] = None):
+        """
+        Function to generate a `pd.DataFrame` of information about each global
+        metadata attribute. The `pd.DataFrame` contains all information in the SWxSOC
+        global attribute schema including:
+
+        - description: (`str`) A brief description of the attribute
+        - default: (`str`) The default value used if none is provided
+        - required: (`bool`) Whether the attribute is required by SWxSOC standards
+
 
         Parameters
         ----------
-        file_name : str
-            The name of the file to load from self.source_dir.
+        attribute_name : `str`, optional, default None
+            The name of the attribute to get specific information for.
+
+        Returns
+        -------
+        info: `pd.DataFrame`
+            A table of information about global metadata.
+
+        Raises
+        ------
+        KeyError: If attribute_name is not a recognized global attribute.
+        """
+        import pandas as pd
```

Comment:
I understand why this import statement is in this function, but I would still prefer it to be at the top of the file and imported at the beginning.

Reply:
Yeah I do a lazy import here to allow Pandas to be an optional dependency. For example with HERMES, I don't want to have Pandas as a required dependency of this package. I actually overwrite this function in the HERMES inheritance tree and use an `astropy.Table` instead of a `DataFrame`

Reply:
I don't really like optional dependencies that are required for certain method calls, I think it adds a lot of confusion for the end user. Personally, I would prefer to have it either be required or not included if it's part of the main code base like this. Libraries only used in testing or development have a much clearer dividing line between "developers" and "users" that means I'm ok with keeping it optional. 

If this method was in a different file, for building documentation, I think I would be ok with the optional dependency in that case, because that falls into developer only use. But since it's part of the main class, I don't really like it.